src/content/en/fundamentals/media/manipulating/files.md

```diff
@@ -0,0 +1,331 @@
+content creators often record audio and video to separate files to begin with.
+If resources come from a consumer-grade device, audio and video will be in the
+same file and you'll need to split them.
+
+In the context of preparing media for the web, demuxing is basically a file
+copying operation where I copy one of the streams to a new file. Using Shaka
+Packager I might, for example, pull out the video stream.
+
+    packager input=myvideo.mp4,stream=video,output=myvideo_video.mp4
+
+Notice that the stream descriptor has an input, an output, and a stream type.
+Despite the mp4 extension this output file wouldn't have any sound. To get an
+audio file, I'd rerun this command with `stream=audio` and output it to a
+different file name. But, you might remember from [Application
+Primers](applications) that Shaka Packager can take multiple stream descriptors.
+This means that Shaka Packager lets me split the audio and video in a single
+command.
+
+    packager \
+      input=myvideo.mp4,stream=video,output=myvideo_video.mp4 \
+      input=myvideo.mp4,stream=audio,output=myvideo_audio.m4a
+
+A full discussion of audio and video formats is beyond the scope of this
+article. One thing to note is that the audio stream of an mp4 file is the
+[advanced audio coding]([https://en.wikipedia.org/wiki/Advanced_Audio_Coding)
+format for which m4a is a common file extension.
+
+Shaka Packager presents demuxing as though you're _extracting_ a stream into a
+new file. It's a little different in ffmpeg, which presents as though you're
+_stripping_ the stream you don't want. With ffmpeg, you need two operations.
+
+    ffmpeg -i myvideo.webm -vcodec copy -an myvideo_video.webm
+    ffmpeg -i myvideo.webm -acodec copy -vn myvideo_audio.webm
+
+Just as with Shaka Packager, we have both an input and an output file. Another
+difference from Shaka Packager is that the streams are identified with flags
+that refer their codecs. The `-vcodec copy` and `-acodec copy` portions of the
+command tell ffmpeg to copy the streams I want while the `-an` and `-vn` flags
+strip the streams I don't want. The keyword copy means I'm moving the streams
+without changing their codecs. 
+
+A list of the
+[demuxing commands](/web/fundamentals/media/manipulating/cheatsheet#demux_split_audio_and_video)
+are provided on the cheat sheet.
+
+## Change the codec
+
+Continuing downward, we arrive at the codec. As stated earlier, a codec is _not_
+the same thing as a container. Two files of the same container type could hold
+data compressed using completely different codecs. The webm format for example
+allows audio to be encoded using either [vorbis](https://en.wikipedia.org/wiki/Vorbis)
+or [opus](https://en.wikipedia.org/wiki/Opus_(audio_format)). To change the
+codec I need ffmpeg.
+
+In the last section I demuxed the audio and video like this:
+
+    ffmpeg -i myvideo.webm -vcodec copy -an myvideo_video.webm
+    ffmpeg -i myvideo.webm -acodec copy -vn myvideo_audio.webm
```

Comment:
WebM can be used for both audio and video.

WebM defines the file container structure, video and audio formats. WebM files consist of video streams compressed with the VP8 or VP9 video codecs and audio streams compressed with the Vorbis or Opus audio codecs.